gallery/src/ThemeSettingsModal/ThemeSettingsModal.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+
+import {
+  NxButton,
+  NxModal,
+  NxFormGroup,
+  NxToggle,
+  NxFieldset,
+  NxRadio,
+  NxTooltip,
+  NxFontAwesomeIcon,
+  NxCode,
+  useToggle
+} from '@sonatype/react-shared-components';
+
+import { faInfoCircle } from '@fortawesome/free-solid-svg-icons';
+
+import { themingEnabled, themeOverride, setThemingEnabled, setThemeOverride } from '../themeClassManager';
+
+type Props = {
+  onClose: () => void;
+};
+
+const ThemeSettingsModal = ({ onClose }: Props) => {
+  const [themingEnabledState, toggleThemingEnabledState] = useToggle(themingEnabled),
+      [themeOverrideState, setThemeOverrideState] = useState(themeOverride);
+
+  useEffect(function() {
+    setThemingEnabled(themingEnabledState);
+  }, [themingEnabledState]);
+
+  useEffect(function() {
+    setThemeOverride(themeOverrideState);
+  }, [themeOverrideState]);
+
+  return (
+    <NxModal id="nx-modal-dark-mode-example"
+             onCancel={onClose}
+             aria-labelledby="nx-modal-dark-mode-header">
+      <header className="nx-modal-header">
+        <h2 className="nx-h2" id="nx-modal-dark-mode-header">
```

Comment:
for consistency I wonder if we should also rename this to something like `gallery-theme-settings-modal-header`

Reply:
> for consistency I wonder if we should also rename this to something like `gallery-theme-settings-modal-header`

Thanks for clarifying about the prefix. Changed here: [af68c48](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/af68c4892214e891dcd143b2b0fa1fc4c71c04e2)